keps/sig-node/3008-qos-class-resources/README.md

```diff
@@ -0,0 +1,2249 @@
+control plane.
+
+For each of these, fill in the following—thinking about running existing user workloads
+and creating new ones, as well as about cluster-level services (e.g. DNS):
+  - [Dependency name]
+    - Usage description:
+      - Impact of its outage on the feature:
+      - Impact of its degraded performance or high-error rates on the feature:
+-->
+
+Implementation Phase 1: A container runtime with support for the new CRI API
+fields is required.
+
+### Scalability
+
+<!--
+For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them.
+
+For beta, this section is required: reviewers must answer these questions.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### Will enabling / using this feature result in any new API calls?
+
+<!--
+Describe them, providing:
+  - API call type (e.g. PATCH pods)
+  - estimated throughput
+  - originating component(s) (e.g. Kubelet, Feature-X-controller)
+Focusing mostly on:
+  - components listing and/or watching resources they didn't before
+  - API calls that may be triggered by changes of some Kubernetes resources
+    (e.g. update of object X triggers new updates of object Y)
+  - periodic API calls to reconcile state (e.g. periodic fetching state,
+    heartbeats, leader election, etc.)
+-->
+
+No, enabling or using the feature does not induce any new API calls in
+Kubernetes.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+<!--
+Describe them, providing:
+  - API type
+  - Supported number of objects per cluster
+  - Supported number of objects per namespace (for namespace-scoped objects)
+-->
+
+Implementation Phase 1: No.
+
+Future implementation phases: QoS-class resources do extend existing API types
+but presumably not introduce new types of objects. However, the design for
+resource discovery and permission control is not ready which might change this.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+<!--
+Describe them, providing:
+  - Which API(s):
+  - Estimated increase:
+-->
+
+No, enabling or using the feature does not result in any new calls to the cloud
+provider.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+<!--
+Describe them, providing:
+  - API type(s):
+  - Estimated increase in size: (e.g., new annotation of size 32B)
+  - Estimated amount of new objects: (e.g., new Object X for every existing Pod)
+-->
+
+Implementation Phase 1: [pod annotations](#pod-annotations) are used as the
+initial user interface so assign QoS-class resources to containers. Exact size
+of each annotation varies (depending on the type of resource) but the
+annotation key is expected to be few tens of bytes. The value part is the name
+of the class expected to be a few bytes long.
```

Comment:
- Are these public annotation keys (starting with a DNS domain name then a slash then an identifier)?
- How would we ever phase these out?

Reply:
Uh-oh, the PRR questionnaire was badly outdated with mentions of the Pod annotation "UX" that has been dropped from the proposal (except for the "Alternatives" section. Thx for commenting this.

@sftim I now updated the whole PRRQ section, no annotations are mentioned there -> I consider this aspect of PRRQ fixed

Reply:
OK, sure.